app/controllers/sessions_controller.rb

```diff
@@ -1,21 +1,36 @@
 class SessionsController < ApplicationController
+  before_action :load_user, only: :create
+
   def new; end
 
   def create
-    user = User.find_by(email: params[:session][:email].downcase)
-    if user&.authenticate(params[:session][:password])
-      flash[:success] = t("flash.login.successed")
-      log_in user
-      redirect_to user
-    else
-      flash.now[:danger] = t("flash.login.failed")
-      render :new
-    end
+    auth
   end
 
   def destroy
     flash[:success] = t("flash.logout")
     log_out
     redirect_to root_url
   end
+
+  private
+
+  def load_user
+    @user = User.find_by(email: params[:session][:email].downcase)
+    return if @user.present?
+
+    flash.now[:danger] = t("flash.login.invalid_email")
+    render :new
+  end
+
+  def auth
+    if @user&.authenticate(params[:session][:password])
```

Comment:
```suggestion
    if @user.authenticate(params[:session][:password])
```
ko cần, vì user bị nil đã xử lý ở trên rồi